src/XMakeBuildEngine/BackEnd/BuildManager/BuildParameters.cs

```diff
@@ -1002,7 +1005,11 @@ private void FindMSBuildExe()
             }
 
             // Use the default location of the directory from which the engine was loaded.
+#if FEATURE_APPDOMAIN
             path = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "MSBuild.exe");
+#else
+            path = Path.GetDirectoryName(Process.GetCurrentProcess().MainModule.FileName);
```

Comment:
I like that philosophy.  We need to figure out how to balance it with not creating too much long-term debt.

(This is mostly because I hit a similar issue and just changed the product code. :innocent:)